src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs

```diff
@@ -2051,7 +2046,7 @@ private void ComputeLayout()
             uint bucketsEstimate = (uint)(_Entries.Count / _nFillFactor);
 
             // Round number of buckets up to the power of two
-            _nBuckets = (uint)(1 << HighestBit(bucketsEstimate));
+            _nBuckets = (uint)HighestBit(bucketsEstimate);
```

Comment:
Should be better to use `RoundUpToPowerOf2`. The old implementation would round up 2^n to next bigger value.

Reply:
I noticed this comment yesterday too, will use that instead.. thanks!

Reply:
Actually, I think the commend here is in accurate that the existing code did not round __up__ to the power of two but only set the highest bit.

e.g.
```csharp
int bucketEstimate = 0b0000_1010;
int highestBitSet = 1 << HighestBit(bucketEstimate); // returns: 0b000_1000
int nextPow2 = BitOperations.RoundUptoPowerOf2(bucketEstimate); // return 0b0001_0000
```

Reply:
Yeah, at a glance it rather looks like the logic for `HighestBit` is incorrect. It claims to be computing the following, where one would normally interpret "rounded up" as  meaning the actual algorithm is `1 + ceil(log2(x)) for non-zero x`
> Returns 1 + log2(x) rounded up, 0 iff x == 0

What's actually happening, however, is that there is no rounding up and it's simply doing `1 + log2(x) for non-zero x`:
```
Console.WriteLine($"{HighestBit(0)} {BitOperations.Log2(0)}");  // 0 0 -- correct
Console.WriteLine($"{HighestBit(1)} {BitOperations.Log2(1)}");  // 1 0 -- correct
Console.WriteLine($"{HighestBit(2)} {BitOperations.Log2(2)}");  // 2 1 -- correct
Console.WriteLine($"{HighestBit(3)} {BitOperations.Log2(3)}");  // 2 1 -- should be 3, as log2(3) is 1.58496...
Console.WriteLine($"{HighestBit(4)} {BitOperations.Log2(4)}");  // 3 2 -- correct
Console.WriteLine($"{HighestBit(5)} {BitOperations.Log2(5)}");  // 3 2 -- should be 4, as log2(5) is 2.321928...
...
```